### PR TITLE
TD-9 add GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by marvinobig**
## This Ticket Includes:
- Tests for status 200 on GET path "/api/articles/:article_id/comments"
- Test for status 400 on GET path "/api/articles/:article_id/comments"
- Test for status 404 on GET path "/api/articles/:article_id/comments"
- Controller, model, route for GET path "/api/articles/:article_id/comments"

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -229,3 +229,52 @@ describe("GET /api/articles", () => {
...
+  });
+  test("status:404, should return error message when id given is not available", () => {
+    return request(app)
+      .get("/api/articles/20/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Comments For Article 20 Not Found");
+      });
+  });
+});
```
Comment (corkesw) [inline __tests__/app.test.js:280]: I'm not clear on what this test is checking. Is it that the article doesn't exist? In which case a 404 is appropriate. Or is it that the article exists but has no comments? If this is the case the I would expect a 200 and an empty array to be retuned form the api.

File: models/api.model.js [hunk 2]
```diff
@@ -59,7 +59,18 @@ exports.fetchArticles = async () => {
...
+exports.fetchArticleCommentsById = async (id) => {
+  const articleComments = await db.query(
+    "SELECT * FROM comments WHERE article_id=$1;",
+    [id]
+  );
+
+  if (articleComments.rows.length === 0) {
+    throw new Error(`Comments For Article ${id} Not Found`, { cause: 404 });
+  }
+
```
Comment (corkesw) [inline models/api.model.js:74]: Think about what we are checking for here as we will get the same response if there is no article as we would if there was an article but no comments. See my comments on the test suite for clarification.

File: __tests__/app.test.js (hunk 1 above)
Comment (marvinobig) [reply __tests__/app.test.js:280]: It's checking that the article exists

Comment (marvinobig) [reply __tests__/app.test.js:280]: I thought we had to check for the ID existing and also why do we need to return an empty array?

Comment (scar1377) [reply __tests__/app.test.js:249]: Don't have to return the empty array, can send a message of "no comments related to this article" , but the code should be 200 instead 404. 'cause it's not an error

File: models/api.model.js (hunk 2 above)
Comment (scar1377) [reply models/api.model.js:74]: change the status from 404 to 200

**Review by scar1377 — APPROVED** [review]
Hi Marv. The code looks fine, just need to change the 404 to 200 for no comments related to the existed article.

**Review by scar1377 — APPROVED** [review]
Good to merge

**Pull request merged**